color/src/color.rs

```diff
@@ -89,17 +89,19 @@ pub struct PremulColor<CS> {
 #[derive(Clone, Copy, Default, Debug, PartialEq)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
 #[non_exhaustive]
+#[repr(u8)]
 pub enum HueDirection {
     /// Hue angles take the shorter of the two arcs between starting and ending values.
     #[default]
-    Shorter,
+    Shorter = 0,
     /// Hue angles take the longer of the two arcs between starting and ending values.
-    Longer,
+    Longer = 1,
     /// Hue angles increase as they are interpolated.
-    Increasing,
+    Increasing = 2,
     /// Hue angles decrease as they are interpolated.
-    Decreasing,
+    Decreasing = 3,
     // It's possible we'll add "raw"; color.js has it.
+    // NOTICE: If a new value is added, be sure to add modify `MAX_VALUE` in the bytemuck impl.
```

Comment:
https://doc.rust-lang.org/stable/core/ptr/fn.read.html is const stable, so it *might* be possible to write a test which fails to compile if this is correct (i.e. using a compile_fail doctest). I don't have time at the moment to prototype this.

Reply:
I'm dumb so I don't even have a clue what you mean.

Reply:
Yeah, if only Rust had a real reflection system on the way.

Reply:
```rust
#[repr(u8)]
enum MyEnum {
    A = 0,
    B = 1
}

const _: MyEnum = {
    let val = 1u8;
    unsafe {
        core::ptr::read((&raw const val).cast())
    }
};
```
compiles, whereas:
```rust
const _: MyEnum = {
    let val = 2u8;
    unsafe {
        core::ptr::read((&raw const val).cast())
    }
};
```
Fails to compile. This is on latest stable.